vite.config.js

```diff
@@ -0,0 +1,162 @@
+import { defineConfig } from 'vite';
+import preact from '@preact/preset-vite';
+import { viteStaticCopy } from 'vite-plugin-static-copy';
+import { VitePWA as vitePWA } from 'vite-plugin-pwa';
+import replace from '@rollup/plugin-replace';
+import yaml from 'yaml';
+import { Feed } from 'feed';
+import config from './src/config.json';
+
+/**
+ * @param {string} content
+ * @param {string} path
+ * @returns {string}
+ */
+function compileMarkdown(content, path) {
+	// Find YAML FrontMatter preceeding a markdown document
+	const FRONT_MATTER_REG = /^\s*---\n\s*([\s\S]*?)\s*\n---\n/i;
+
+	// Find a leading title in a markdown document
+	const TITLE_REG = /^\s*#\s+(.+)\n+/;
+
+	const matches = content.match(FRONT_MATTER_REG);
+	if (matches) {
+		let meta;
+		try {
+			meta =
+				yaml.parse('---\n' + matches[1].replace(/^/gm, '  ') + '\n') ||
+				{};
+		} catch (e) {
+			throw new Error(`Error parsing YAML FrontMatter in ${path}`);
+		}
+		content = content.replace(FRONT_MATTER_REG, '');
+		if (!meta.title) {
+			let [, title] = content.match(TITLE_REG) || [];
+			if (title) {
+				meta.title = title;
+			}
+		}
+
+		content = '---\n' + JSON.stringify(meta) + '\n---\n' + content;
+	}
+
+	return content;
+}
+
+export default defineConfig({
+	publicDir: 'src/assets',
+	build: {
+		target: ['chrome88', 'edge88', 'es2020', 'firefox78', 'safari14'],
```

Comment:
Default is this list but `chrome87`, not 88.

Unfortunately Vite's CSS transpiler cannot handle a few of our nested classes without generating an `is:(...)`, which is only available in chrome88. The tiny bump from defaults felt warranted, though we can discuss other targets. Might need some refactoring is all.